src/net.cpp

```diff
@@ -970,7 +970,6 @@ void CConnman::AcceptConnection(const ListenSocket& hListenSocket) {
     CAddress addr;
     int nInbound = 0;
     int nMaxInbound = nMaxConnections - (nMaxOutbound + nMaxFeeler);
```

Comment:
I disagree at just blindly capping this at 0. There are only two ways imo:

* Add a comment which says this may be negative
* Rework the "-maxconnections" code to not allow negative numbers here